src/box/memtx_engine.cc

```diff
@@ -1354,10 +1354,38 @@ memtx_join_f(va_list ap)
 	return rc;
 }
 
+#if defined(ENABLE_CHECKPOINT_JOIN)
+#include "memtx_checkpoint_join.c"
+#else /* !defined(ENABLE_CHECKPOINT_JOIN) */
+
+#define memtx_engine_checkpoint_join generic_engine_join
+
+int
+memtx_engine_recover_synchro(struct memtx_engine *memtx,
```

Comment:
AFAICS this function isn't used anywhere in CE so we don't need a stub for it.

Reply:
But it's used in EE. It just seems not right to put such function in `relay_checkpoint_join.cc`. Morever, as it does pretty the same work as `memtx_engine_checkpoint_join`, I made a common part for them, which recovers snap in the separate thread. If I put it in `relay_checkpoint_join.cc` then it'll be a major code duplication. I don't like that a lot